src/runtime.rs

```diff
@@ -374,12 +421,9 @@ struct SharedState<DB: Database> {
     /// (Ideally, this should be `AtomicU64`, but that is currently unstable.)
     revision: AtomicUsize,
 
-    /// Counts the number of pending increments to the revision
-    /// counter. If this is non-zero, it means that the current
-    /// revision is out of date, and hence queries are free to
-    /// "short-circuit" their results if they learn that. See
-    /// `is_current_revision_canceled` for more information.
-    pending_revision_increments: AtomicUsize,
+    /// If revision N is the last revision to be cancelled, then this
+    /// stores N+1. Stores 0 if there have been no cancellations.
+    next_uncanceled_revision: AtomicUsize,
```

Comment:
Hm, I am slightly confused here... Am I correct that `next_uncanceled_revision` may be *earlier* then revision? The `next_` word sort-of makes me think that this is necessary a revision in the future. Perhaps flip this to `last_cancelled_revision`?

Reply:
I debated a lot about what to name it. `last_canceled_revision` isn't quite right, because it stores N+1 (where N is the last canceled revision). That's important because otherwise the initial value would suggest that R0 was canceled. I could make R0 not a valid revision (and just start the revisions at R1). Maybe I should just do that.

Reply:
Making initial revision 1 seems natural.